CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java

```diff
@@ -3,128 +3,64 @@
 import com.cedarpolicy.Experimental;
 import com.cedarpolicy.ExperimentalFeature;
 import com.cedarpolicy.model.AuthorizationResponse.Decision;
-import com.cedarpolicy.model.AuthorizationResponse.Diagnostics;
-import com.cedarpolicy.model.slice.Policy;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.JsonNode;
-import com.google.common.collect.ImmutableSet;
 
-import java.util.List;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
 
 @Experimental(ExperimentalFeature.PARTIAL_EVALUATION)
-public abstract class PartialAuthorizationResponse {
-    private final Diagnostics diagnostics;
+public class PartialAuthorizationResponse {
+    private Decision decision;
+    private Set<String> satisfied;
+    private Set<String> errored;
+    private Set<String> may_be_determining;
+    private Set<String> must_be_determining;
+    private Map<String, JsonNode> residuals;
+    private Set<String> nontrivial_residuals;
+
+    public PartialAuthorizationResponse(Decision decision, Set<String> satisfied, Set<String> errored,
+            Set<String> may_be_determining, Set<String> must_be_determining, Map<String, JsonNode> residuals,
+            Set<String> nontrivial_residuals) {
+        this.decision = decision;
+        this.errored = new HashSet<String>(errored);
+        this.may_be_determining = new HashSet<String>(may_be_determining);
+        this.must_be_determining = new HashSet<String>(must_be_determining);
+        this.residuals = new HashMap<String, JsonNode>(residuals);
+        this.nontrivial_residuals = new HashSet<String>(nontrivial_residuals);
+    }
+
+    public Decision getDecision() {
+        return this.decision;
+    }
```

Comment:
should this have docs, eg, noting what returning `null` means?

Reply:
Done.